\ref{} is detected as end of a sentence

When I use `\ref{}`, the command was detected as end of a sentence:  
![image](https://user-images.githubusercontent.com/68371029/119588090-043daa80-be0b-11eb-99ac-fa380a165ba0.png)

It is expected to detect an error in sentences such as line 199, but errors outed in line 200 and 201.

`\ref{}` command is used for cross-reference frequently in academic writing.  
In fact, such structure of sentences were contained a lot in my file, so the output were filled with these errors.

## How to reproduce

### Environment

- Node.js v14.17.0
- textlint-plugin-latex2e v1.1.3
- textlint v12.0.0
- textlint-rule-ja-no-mixed-period v2.1.1

### Procedure

```bash
$ mkdir textlint-latex2e-issue168 && cd $_
$ npm init -y
$ npm install --save \
    textlint@12.0.0 \
    textlint-plugin-latex2e@1.1.3 \
    textlint-rule-ja-no-mixed-period@2.1.1
$ cat << EOF > sample.tex
本日は図\ref{fig:sunnyy}に示すように晴天なり。
EOF
$ cat << EOF > .textlintrc
{
    "plugins": ["latex2e"],
    "rules": {
        "ja-no-mixed-period": {
            "periodMark": "。"
        }
    }
}
EOF
$ textlint sample.tex

/Users/pudding/Factory/textlint-latex2e-issue168/sample.tex
  1:4  error  文末が"。"で終わっていません。  ja-no-mixed-period

✖ 1 problem (1 error, 0 warnings)
```

## Cause of this issue

There seems to be an error in the way `Paragraph` is interpreted. There should be an inline node and a block node, but since `\ref` is interpreted as a block node, there are two separate Paragraphs, which should be one.

```
$ npx tex2tast -r sample.tex
{
  "type": "Document",
  "raw": "本日は図\\ref{fig:sunnyy}に示すように晴天なり。\n",
  "children": [
    {
      "raw": "本日は図",
      "type": "Paragraph",
      "children": [
        {
          "raw": "本日は図",
          "type": "Str",
          "value": "本日は図"
        }
      ]
    },
    {
      "raw": "\\ref{fig:sunnyy}",
      "value": "fig:sunnyy",
      "type": "Html"
    },
    {
      "raw": "に示すように晴天なり。",
      "type": "Paragraph",
      "children": [
        {
          "raw": "に示すように晴天なり。",
          "type": "Str",
          "value": "に示すように晴天なり。"
        }
      ]
    }
  ]
}
```

This problem can be avoided by interpreting it as an inline element, just like label and url.

--- a/src/paragraphize.ts
+++ b/src/paragraphize.ts
@@ -39,7 +39,7 @@ export default (rootNode: TxtParentNode): TxtParentNode => {
     switch (node.type) {
       case ASTNodeTypes.CodeBlock:
       case ASTNodeTypes.Header:
-      case ASTNodeTypes.Html:
+      case ASTNodeTypes.HtmlBlock:
         pushChild(paragraph);
         paragraph = [];
         children.push(node);

--- a/test/latex-to-ast.test.ts
+++ b/test/latex-to-ast.test.ts
@@ -210,9 +210,10 @@ C`;
     const code = `\\ref{label}`;
     const actual = parse(code);
     expect(actual.children.length).toBe(1);
-    expect(actual.children[0].type).toBe(ASTNodeTypes.Html);
-    expect(actual.children[0].value).toBe("label");
-    expect(actual.children[0].raw).toBe(code);
+    expect(actual.children[0].type).toBe(ASTNodeTypes.Paragraph);
+    expect(actual.children[0].children[0].type).toBe(ASTNodeTypes.Html);
+    expect(actual.children[0].children[0].value).toBe("label");
+    expect(actual.children[0].children[0].raw).toBe(code);
   });
   test("CodeBlock is not to be included in paragraph", () => {
     const code = `\\section{title}
@@ -227,6 +228,12 @@ fugafuga`;
     expect(actual.children[2].type).toBe(ASTNodeTypes.Paragraph);
     expect(actual.children[4].type).toBe(ASTNodeTypes.CodeBlock);
     expect(actual.children[6].type).toBe(ASTNodeTypes.Paragraph);
+  });
+  test("ref should be included in paragraph", () => {
+    const code = `This sentence \\ref{refs} must be parsed as one paragraph.`;
+    const actual = parse(code);
+    expect(actual.children.length).toBe(1);
+    expect(actual.children[0].type).toBe(ASTNodeTypes.Paragraph);
   });
 });
 

